packages/thirdweb/src/wallets/aws-kms.ts

```diff
@@ -0,0 +1,129 @@
+import type { KMSClientConfig } from "@aws-sdk/client-kms";
+import { KmsSigner } from "aws-kms-signer";
+import type {
+  SignableMessage,
+  TransactionSerializable,
+  TypedData,
+  TypedDataDefinition,
+} from "viem";
+import { hashTypedData, toBytes } from "viem";
+import { getCachedChain } from "../chains/utils.js";
+import type { ThirdwebClient } from "../client/client.js";
+import { eth_sendRawTransaction } from "../rpc/actions/eth_sendRawTransaction.js";
+import { getRpcClient } from "../rpc/rpc.js";
+import { serializeTransaction } from "../transaction/serialize-transaction.js";
+import type { Address } from "../utils/address.js";
+import type { Hex } from "../utils/encoding/hex.js";
+import { keccak256 } from "../utils/hashing/keccak256.js";
+import type { Account } from "./interfaces/wallet.js";
+
+const Buffer = globalThis.Buffer;
+
+type SendTransactionResult = {
+  transactionHash: Hex;
+};
+
+type SendTransactionOption = TransactionSerializable & {
+  chainId: number;
+};
+
+type AwsKmsAccountOptions = {
+  keyId: string;
+  config?: KMSClientConfig;
+  client: ThirdwebClient;
+};
+
+export async function getAwsKmsAccount(
+  options: AwsKmsAccountOptions,
+): Promise<Account> {
+  if (typeof Buffer === "undefined") {
+    throw new Error("AwsKmsAccount only works in Node.js environment");
+  }
+
+  const { keyId, config, client } = options;
+  const signer = new KmsSigner(keyId, config);
+
+  // Populate address immediately
+  const addressUnprefixed = await signer.getAddress();
+  const address = `0x${addressUnprefixed}` as Address;
+
+  async function signTransaction(tx: TransactionSerializable): Promise<Hex> {
+    const serializedTx = serializeTransaction({ transaction: tx });
+    const txHash = keccak256(serializedTx);
+
+    // we don't polyfill buffer, but signer.sign explicitly requires a buffer
+    // what do we do here?
```

Comment:
Was this resolved with your check above?

Reply:
yes! removing comments